service/authorization/v2/authorization.go

```diff
@@ -180,8 +185,60 @@ func (as *Service) GetDecisionMultiResource(ctx context.Context, req *connect.Re
 }
 
 // GetDecisionBulk for multiple requests, each comprising a combination of entity chain, action, and one or more resources
-func (as *Service) GetDecisionBulk(_ context.Context, _ *connect.Request[authzV2.GetDecisionBulkRequest]) (*connect.Response[authzV2.GetDecisionBulkResponse], error) {
-	return nil, connect.NewError(connect.CodeUnimplemented, errors.New("GetDecisionBulk not implemented"))
+func (as *Service) GetDecisionBulk(ctx context.Context, req *connect.Request[authzV2.GetDecisionBulkRequest]) (*connect.Response[authzV2.GetDecisionBulkResponse], error) {
+	as.logger.DebugContext(ctx, "getting decision bulk")
+
+	ctx, span := as.Tracer.Start(ctx, "GetDecisionBulk")
+	defer span.End()
+
+	// Extract trace context from the incoming request
+	propagator := otel.GetTextMapPropagator()
+	ctx = propagator.Extract(ctx, propagation.HeaderCarrier(req.Header()))
+
+	pdp, err := access.NewJustInTimePDP(ctx, as.logger, as.sdk)
+	if err != nil {
+		as.logger.ErrorContext(ctx, "failed to create JIT PDP", slog.Any("error", err))
+		return nil, connect.NewError(connect.CodeInternal, err)
+	}
+
+	multiRequests := req.Msg.GetDecisionRequests()
+	decisionResponses := make([]*authzV2.GetDecisionMultiResourceResponse, len(multiRequests))
+
+	// TODO: revisit performance of this loop after introduction of caching and registered resource values within decisioning,
+	// as the same entity in multiple requests should only be resolved JIT once, not once per request if the same in each.
+	for idx, request := range multiRequests {
+		entityIdentifier := request.GetEntityIdentifier()
+		action := request.GetAction()
+		resources := request.GetResources()
+
+		decisions, allPermitted, err := pdp.GetDecision(ctx, entityIdentifier, action, resources)
```

Comment:
Nit: would rename this to `decision` since the method is singular and so is the `rollupMultiResourceDecision` method